docs/DeveloperGuide.md

```diff
@@ -809,17 +811,84 @@ There are -FLAGS for the various lessons options per module:
 ____________________________________________________________
 ```
 #### Adding a ToDo/Event/Deadline
+Prerequisite: Make sure you are in the main interface.
+#### Invalid Commands
+1. Test case for empty task description: ```todo ``` or ```deadline``` or ```event```
+Expected: Exception thrown. Error details shown in status message
+2. Test case for invalid formats
+   - Out of calendar range: ```deadline return book -by 40-11-2023-11:23``` or ```event wedding -from 40-11-2023-11:23 -to 41-11-2023-11:23```
+   - Invalid dateTime format ```deadline return book -by 2023-10-11-11:23``` or ```event wedding -from 2023-10-11-11:23 -to 2023-10-12-11:23```
+   - Missing parameters ``` deadline return book 15-11-2023-11:23``` or ```event wedding 16-11-2023-11:23 -to 20-11-2023-11:23```
+   - Extra parameters ```deadline return book -by 17-11-2023-11:23 blah blah```
```

Comment:
Looks good, I think you can add in one more case for events/deadline that occur before system date time